src/query/api/v1/handler/prom/prom.go

```diff
@@ -0,0 +1,127 @@
+package prom
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/m3db/m3/src/query/api/v1/handler/prometheus/native"
+	"github.com/m3db/m3/src/query/api/v1/handler/prometheus/remote"
+	"github.com/m3db/m3/src/query/api/v1/options"
+	"github.com/m3db/m3/src/query/storage/prometheus"
+	"github.com/prometheus/prometheus/promql"
+)
+
+const prometheusHandlerPrefix = ""
+
+// Options areoptions for the prometheus read handlers.
+type Options struct {
+	PromQLEngine *promql.Engine
+}
+
+// NewPrometheusReadHandlers creates new prometheus query and query_range
+// handlers.
+func NewPrometheusReadHandlers(opts Options) ([]options.CustomHandler, error) {
+	return []options.CustomHandler{
```

Comment:
This should not return `CustomHandler`s.. Those are meant for extensions to the m3 query engine.. We should create these prometheus endpoints akin to the regular endpoints as we do [here](https://github.com/m3db/m3/blob/master/src/query/api/v1/httpd/handler.go#L121)